### ISSUE TITLE
Getting error that domain must be fully qualified

### ISSUE THREAD

**Issue by deberon** [terraform-providers/terraform-provider-dns#19]
When trying to apply the example given at https://www.terraform.io/docs/providers/dns/r/dns_a_record_set.html I receive the following error: 

dns_a_record_set.www: Error updating DNS record: dns: domain must be fully qualified

### Terraform Version
```
$ terraform -v
Terraform v0.10.8
```

### Affected Resource(s)
Please list the resources as a list, for example:
- dns_ptr_record
- dns_aaaa_record_set
- dns_cname_record
- dns_a_record_set

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
provider "dns" {
    update {
        server = "REDACTED"
        key_name = "terraform"
        key_algorithm = "hmac-md5"
        key_secret = "REDACTED"
    }
}

resource "dns_ptr_record" "dns-sd" {
  zone = "example.com."
  name = "r._dns-sd"
  ptr  = "example.com."
  ttl  = 300
}

/*resource "dns_aaaa_record_set" "www" {
  zone = "example.com."
  name = "www"
  addresses = [
    "fdd5:e282:43b8:5303:dead:beef:cafe:babe",
    "fdd5:e282:43b8:5303:cafe:babe:dead:beef",
  ]
  ttl = 300
}*/

/*resource "dns_cname_record" "foo" {
  zone  = "example.com."
  name  = "foo"
  cname = "bar.example.com."
  ttl   = 300
}*/

/*resource "dns_a_record_set" "www" {
  zone = "example.com."
  name = "www"
  addresses = [
    "192.168.0.1",
    "192.168.0.2",
    "192.168.0.3",
  ]
  ttl = 300
}*/
```

### Expected Behavior
Record should have been created.

### Actual Behavior
Received error that the domain was not fully qualified, even though it was.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Copy above config into main.tf and replace proper credentials
2. `terraform apply`

### Important Factoids
It's not just the examples. When I tried adding my own records I receive the same error.

### References
- GH-13095: I tried the fix in this issue, but got an authentication error instead.

**Comment by deberon** [comment]
I am also using v1.0.0 of the DNS provider.

**Comment by deberon** [comment]
Anybody around to take a look at this? From what I can see the basic functionality of this tool is broken. I can't even recreate the examples given on the website.

**Comment by deberon** [comment]
It turns out this issue was a combination of whatever bug this is: GH-13095 and an error in my DNS configuration. The error provided could definitely be a little more helpful.

**Comment by apparentlymart** [comment]
Hi @deberon! Sorry for the delayed response here. With the 0.11.0 release followed by the U.S. holidays and then some staff illness we've got a little behind here. :confounded:

I think the issue you're referencing in your latest comment is hashicorp/terraform#13095. Am I understanding correctly that you got your config working by following the advice in that issue and adding a dot to the end of the `key_name` in the provider configuration? If so, I can look into what's going on there and at least produce a better error message and update the doc examples, as you mentioned.

**Comment by deberon** [comment]
Yep that is correct. I had initially added the dot at the end but had an error in my bind configuration and incorrectly assumed I was moving backwards in my troubleshooting so I reverted my change. Thanks!

**Comment by katbyte** [comment]
Should be fixed by #36
